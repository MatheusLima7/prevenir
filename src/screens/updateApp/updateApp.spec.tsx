import { render } from "@testing-library/react-native";

import UpdateScreen from "./updateApp";

jest.mock("react-native-linear-gradient", () => jest.fn());

jest.mock("@react-native-vector-icons/FontAwesome6", () => {
  return (props: any) => `MockedIcon(${JSON.stringify(props)})`;
});

jest.mock("@react-native-vector-icons/evil-icons", () => {
  return (props: any) => `MockedIcon(${JSON.stringify(props)})`;
});

jest.mock("react-redux", () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

const mockedNavigate = jest.fn();

jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

describe("Update Screen Component", () => {
  it("should render component correctly", () => {
    const container = render(<UpdateScreen />);

    expect(container).toMatchSnapshot();
  });
});
