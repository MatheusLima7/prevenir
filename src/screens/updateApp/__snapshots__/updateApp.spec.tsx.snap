// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update Screen Component should render component correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "backgroundColor": "#ECFEFF",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "space-between",
      "paddingBottom": 50,
      "paddingLeft": 30,
      "paddingRight": 30,
      "paddingTop": 25,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/assets/prevenir.png",
      }
    }
    style={
      {
        "height": 83,
        "width": 190,
      }
    }
  />
  <View
    style={{}}
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 35,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/cellphone.png",
          }
        }
        style={{}}
      />
      <Text
        color="gray"
        size="xLarge"
        style={
          [
            {
              "color": "#717171",
              "display": "flex",
              "fontFamily": "Quicksand-SemiBold",
              "fontSize": 30,
              "fontWeight": "600",
              "lineHeight": 30,
            },
            {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginLeft": 20,
            },
          ]
        }
        testId="text-component"
        type="title"
      >
        Atualize seu aplicativo!
      </Text>
    </View>
    <Text
      color="gray"
      size="small"
      style={
        [
          {
            "color": "#717171",
            "display": "flex",
            "fontFamily": "Quicksand-SemiBold",
            "fontSize": 14,
            "fontWeight": "600",
            "lineHeight": 20,
          },
          {
            "marginBottom": 25,
          },
        ]
      }
      testId="text-component"
      type="title"
    >
      Voc√™ est√° usando uma vers√£o desatualizada deste aplicativo. üò¥ Bora atualizar?
    </Text>
    <Text
      color="gray"
      size="small"
      style={
        [
          {
            "color": "#717171",
            "display": "flex",
            "fontFamily": "Quicksand-Regular",
            "fontSize": 14,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          {},
        ]
      }
      testId="text-component"
      type="text"
    >
      A vers√£o do aplicativo instalada no seu smartphone est√° desatualizada. Para seguir, atualize o aplicativo ou clique no bot√£o 
      <Text
        color="gray"
        size="small"
        style={
          [
            {
              "color": "#717171",
              "display": "flex",
              "fontFamily": "Quicksand-SemiBold",
              "fontSize": 14,
              "fontWeight": "600",
              "lineHeight": 20,
            },
            {},
          ]
        }
        testId="text-component"
        type="title"
      >
        "
        Atualizar agora
        "
      </Text>
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItens": "center",
        "borderBottomLeftRadius": 7,
        "borderBottomRightRadius": 7,
        "borderTopLeftRadius": 7,
        "borderTopRightRadius": 7,
        "color": "#717171",
        "display": "flex",
        "flexDirection": "row",
        "fontSize": 14,
        "justifyContent": "space-evenly",
        "lineHeight": 20,
        "opacity": 1,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "width": "auto",
      }
    }
  />
</RNCSafeAreaView>
`;
